src/test/java/com/yegor256/tojos/MnSynchronizedTest.java

```diff
@@ -0,0 +1,193 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.yegor256.tojos;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link MnSynchronized}.
+ *
+ * @since 0.3.0
+ */
+class MnSynchronizedTest {
+
+    /**
+     * Number of threads.
+     */
+    static final int THREADS = 5;
+
+    /**
+     * The accumulator that contains a changes in the under test mono.
+     */
+    private static Collection<Collection<Map<String, String>>> changes;
+
+    /**
+     * The mono under test.
+     */
+    private Mono mono;
+
+    /**
+     * The executor.
+     */
+    private ThreadPoolExecutor executor;
+
+    /**
+     * The blocking queue.
+     */
+    private BlockingQueue<Runnable> queue;
+
+    /**
+     * The row.
+     */
+    private Map<String, String> row;
+
+    @BeforeEach
+    final void setUp() {
+        this.mono = new MnSynchronized(new MnMemory());
+        MnSynchronizedTest.changes = Collections.synchronizedList(new ArrayList<>(0));
+        this.row = Collections.synchronizedMap(new HashMap<>(0));
+        this.queue = new LinkedBlockingQueue<>(MnSynchronizedTest.THREADS);
+        this.executor = new ThreadPoolExecutor(
+            MnSynchronizedTest.THREADS,
+            MnSynchronizedTest.THREADS,
+            5L,
+            TimeUnit.SECONDS,
+            this.queue,
+            new ThreadPoolExecutor.CallerRunsPolicy()
+        );
+    }
+
+    /**
+     * Thread-safety test.
+     * In this test, we check the number of changes in MnSynchronized mono.
+     * It should be equal to the sum of the arithmetic progression over the number of threads.
+     *
+     * @throws InterruptedException When fails
+     */
+    @Test
+    final void concurrentScenario() throws InterruptedException {
+        this.executor.prestartAllCoreThreads();
+        for (int trds = 0; trds < MnSynchronizedTest.THREADS; ++trds) {
+            this.row.put(Tojos.KEY, String.valueOf(trds));
+            if (!this.queue.offer(new MnSynchronizedTest.TestTask(trds, this.mono, this.row))) {
+                throw new IllegalStateException("Can't put runnable test");
```

Comment:
Do we need to check it? Maybe it's better just omit that check? What do you think?